cve_bin_tool/checkers/kerberos.py

```diff
@@ -20,9 +20,9 @@ def guess_krb5_version_from_content(lines):
     # Library signature looks like "KRB5_BRAND: krb5-1.15.1-final 1.15.1 20170302"
     signatures = [r"KRB5_BRAND: krb5-(\d+\.\d+\.?\d?)-final"]
 
-    new_guess = regex_find(lines, *signatures)
-    if new_guess:
-        return new_guess
+    match = regex_find(lines, *signatures)
+    if match != "UNKNOWN":
+        return match
```

Comment:
We actually *want* this to return UNKNOWN because it'll make the parsing in cli.py more consistent, so this change is bad.  If you want to simplify things, though, you can probably return(regex_find(lines, *signatures)) and it should do the right thing.

Reply:
@terriko I want to discuss a dry run of the whole function. 
```
def guess_krb5_version_from_content(lines):
    """Guesses the krb version from the file contents
    """
    new_guess = "" #new guess is empty string

    # Library signature looks like "KRB5_BRAND: krb5-1.15.1-final 1.15.1 20170302"
 
    signatures = [r"KRB5_BRAND: krb5-(\d+\.\d+\.?\d?)-final"]
    
    #match contains version info from the regex_find
    match = regex_find(lines, *signatures)

    #if the regex_find found a version we can return that, else we can proceed
    if match != "UNKNOWN":
        return match
    
    #we will retry to find the version now with pattern1
    # CLI signature
    pattern1 = re.compile(r"kerberos 5[_-](appl-)*(1+\.[0-9]+(\.[0-9]+)*)")

    for line in lines:
        match = pattern1.search(line)
        if match:
            new_guess2 = match.group(0).strip()
            if len(new_guess2) > len(new_guess):
                new_guess = new_guess2
    #proposed change goes here
    if len(new_guess > 0):
        return new_guess[9:]
    else:
        return "UNKNOWN"
```

I think we must return UNKNOWN if the second signature i.e pattern1 also doesn't find anything. So I propose the following changes. Correct me If I'm wrong.

Reply:
Nope, you're right.  I'd forgotten that this function had two sets of matching mechanics.  Good catch!

Reply:
@terriko Should I update the Kerberos.py file with the proposed changes